integration-tests/src/test_loop/builder.rs

```diff
@@ -36,18 +37,29 @@ use near_vm_runner::{ContractRuntimeCache, FilesystemContractRuntimeCache};
 use nearcore::state_sync::StateSyncDumper;
 use tempfile::TempDir;
 
-use super::env::{TestData, TestLoopEnv};
+use super::env::{ClientToShardsManagerSender, TestData, TestLoopChunksStorage, TestLoopEnv};
 
 pub struct TestLoopBuilder {
     test_loop: TestLoopV2,
     genesis: Option<Genesis>,
     clients: Vec<AccountId>,
+    /// Will store all chunks produced within the test loop.
+    chunks_storage: Arc<Mutex<TestLoopChunksStorage>>,
+    /// Whether test loop should drop all chunks validated by the given account.
+    drop_chunks_validated_by: Option<AccountId>,
```

Comment:
can we collect such parameters into another container, such as AdversarialBehaviorParams/Config or something? I assume there will be more such behaviors added in the future.

Reply:
Yeah... I intentionally didn't introduce container, because if this won't be used by anyone else, this will be useless abstraction (kinda like we have `challenges` field in block which no one uses).
If you have strong opinion on it, I'll add it.

Reply:
for now we can keep separately and later combine into a container, do not have strong opinion on it.